engine/metrics.go

```diff
@@ -0,0 +1,43 @@
+package engine
+
+import (
+	rpcmetrics "github.com/polymerdao/monomer/metrics"
+)
+
+const (
+	MetricsSubsystem = "engine"
+
+	ForkchoiceUpdatedV3MethodName = "forkchoiceUpdatedV3"
+	GetPayloadV3MethodName        = "getPayloadV3"
+	NewPayloadV3MethodName        = "newPayloadV3"
+)
+
+var RPCMethodDurationBucketsMicroseconds = []float64{1, 10, 50, 100, 500, 1000}
+
+// Metrics contains metrics collected from the engine package.
+type Metrics interface {
+	rpcmetrics.Metrics
+}
+
+type metrics struct {
+	rpcmetrics.RPCMetrics
+}
+
+func NewMetrics(namespace string) Metrics {
+	return &metrics{
+		rpcmetrics.NewRPCMetrics(
+			namespace,
+			MetricsSubsystem,
+			"Duration of each engine RPC method call in microseconds",
+			RPCMethodDurationBucketsMicroseconds,
+		),
+	}
+}
```

Comment:
```suggestion
func NewMetrics(namespace string) Metrics {
	rpcm := rpcmetrics.NewRPCMetrics(
		namespace,
		MetricsSubsystem,
		"Duration of each engine RPC method call in microseconds",
		RPCMethodDurationBucketsMicroseconds,
	)
	return &rpcm
}
```

Seems to work. Is there a specific need for the indirection here?

Reply:
Yeah I kept going back and forth on this. If we eventually want to add other metrics to the eth or engine packages outisde of the shared RPC metrics then we'd have to implement them individually in the `eth/metrics.go` or `engine/metrics.go` file and could then easily add them here by adding a line to the metrics struct. However, if we only anticipate having the shared RPC metrics then your suggested change would be fine. Can you think of any other metrics that we may want to eventually add for either of these packages outside of the RPC method call/duration metrics?

cc @joshklop